subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/BuildModelActionRunner.java

```diff
@@ -37,6 +37,7 @@
 
 public class BuildModelActionRunner implements BuildActionRunner {
     @Override
+    @SuppressWarnings("deprecation")
```

Comment:
Looks like not all invocations of  `setConfigureOnDemand` and `isConfigureOnDemand` are annotated with `SuppressWarnings`:

https://github.com/gradle/gradle/blob/a784b7511b82bd8acacc4c4751fd85a6de8734fe/subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ClientProvidedBuildActionRunner.java#L47

https://github.com/gradle/gradle/blob/f727cfb1a8edc6564ec0556be8d41e9c7d92718a/subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuildFactory.java#L112

https://github.com/gradle/gradle/blob/a784b7511b82bd8acacc4c4751fd85a6de8734fe/subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptionFactory.java#L251

https://github.com/gradle/gradle/blob/f2af2a3f0dc03ec0be205c5689589591bc4fb23f/subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java#L325